hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmLCRule.java

```diff
@@ -0,0 +1,239 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.helpers;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.ozone.om.exceptions.OMException;
+
+/**
+ * A class that encapsulates lifecycle rule.
+ */
+public class OmLCRule {
+
+  public static final int LC_ID_LENGTH = 48;
+  // Ref: https://docs.aws.amazon.com/AmazonS3/latest/userguide/intro-lifecycle-rules.html#intro-lifecycle-rule-id
+  public static final int LC_ID_MAX_LENGTH = 255;
+
+  private String id;
+  private String prefix;
+  private boolean enabled;
+  private boolean isPrefixEnable;
+  private boolean isTagEnable;
+  // List of actions for this rule
+  private List<OmLCAction> actions;
+  private OmLCFilter filter;
+
+  OmLCRule(String id, String prefix, boolean enabled,
+      List<OmLCAction> actions, OmLCFilter filter) {
+    this.id = id;
+    this.prefix = prefix;
+    this.enabled = enabled;
+    this.actions = actions;
+    this.filter = filter;
+
+    if (StringUtils.isEmpty(this.id)) {
+      this.id = RandomStringUtils.randomAlphanumeric(LC_ID_LENGTH);
+    }
+
+    OmLifecycleRuleAndOperator andOperator = filter != null ? filter.getAndOperator() : null;
+
+    if (prefix != null ||
+        (filter != null && filter.getPrefix() != null) ||
+        (andOperator != null && andOperator.getPrefix() != null)) {
+      isPrefixEnable = true;
+    }
+
+    if ((filter != null && filter.getTag() != null) ||
+        (andOperator != null && !andOperator.getTags().isEmpty())) {
+      isTagEnable = true;
+    }
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public void setId(String id) {
+    this.id = id;
+  }
+
+  public String getPrefix() {
+    return prefix;
+  }
+
+  public void setPrefix(String prefix) {
+    this.prefix = prefix;
+  }
+
+  public boolean isEnabled() {
+    return enabled;
+  }
+
+  public void setEnabled(boolean enabled) {
+    this.enabled = enabled;
+  }
+
+  public List<OmLCAction> getActions() {
+    return actions;
+  }
+
+  public void setActions(List<OmLCAction> actions) {
+    this.actions = actions;
+  }
+
+  /**
+   * Get the expiration action if present.
+   * This method is for backward compatibility.
+   *
+   * @return the expiration action if present, null otherwise
+   */
+  public OmLCExpiration getExpiration() {
+    if (actions == null || actions.isEmpty()) {
```

Comment:
If at least one action is required, we can skip this check logic.

Reply:
This part just some redundant checks, if the user do not call the `valid` method, they can create OMLCRule with any content

Reply:
I thought the code logic will call valid automatically, instead of user call valid.  Not exactly?

Reply:
Add valid when build, so the valid will be called automatically